tests/app/main/test_suppliers.py

```diff
@@ -712,22 +723,54 @@ def test_shows_supplier_info(self, data_api_client):
             )
 
             for property_str in (
-                "Supplier Description",
-                "Client One",
-                "Client Two",
-                "1 Street",  # "2 Building" is not shown, even though it's in the supplier's contactInformation
-                "supplier.dmdev",
-                "supplier@user.dmdev",
-                "Supplier Person",
-                "0800123123",
-                "Supplierville",
-                #  "Supplierland" is not shown, even though it's in the supplier's contactInformation
-                "11 AB",
+                # "Supplier details" section at the top
+                "Supplier Person",  # Contact name
+                "supplier@user.dmdev",  # Contact email
+                "0800123123",  # Phone number
+                "1 Street",  # Address: "2 Building" and "Supplierland" are not shown, even though in contactInformation
+                "Supplierville",  # Town or City
+                "11 AB",  # Postcode
+                "Supplier Description",  # Supplier summary
+                # "Registration information" section
+                "Official Name Inc",  # Registered company name
+                "CH123456",  # Companies House number
+                "987654321",  # DUNS number
+                "12345678",  # VAT number
+                "Open for business",  # Trading status
+                "Small",  # Size
             ):
                 assert document.xpath("//*[normalize-space(string())=$t]", t=property_str), property_str
 
+            # Registration country and registration number not shown if Companies House ID exists
+            for property_str in ("bz", "BEL153",):
+                assert property_str not in page_html
+
             data_api_client.get_supplier.assert_called_once_with(1234)
 
+    def test_shows_overseas_supplier_info_if_no_companies_house_number(self, data_api_client):
+        data_api_client.get_supplier.return_value = get_supplier(companiesHouseNumber=None)
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.get("/suppliers/details")
+            assert res.status_code == 200
+            page_html = res.get_data(as_text=True)
+            document = html.fromstring(page_html)
+            for property_str in ("BZ", "BEL153",):
+                assert document.xpath("//*[normalize-space(string())=$t]", t=property_str), property_str
```

Comment:
Ah clever remembering to add `property_str` as the message - I should do this more.